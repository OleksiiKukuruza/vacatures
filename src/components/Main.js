import React from 'react';
import styled from 'styled-components';
import Order from './Order';
import ProductList from './ProductList';

const StyledMain = styled.div`
  flex: 1;
  display: flex;
  margin: 0 10%;
`;

const Main = () => (
  <StyledMain>
    <ProductList />
    <Order />
  </StyledMain>
);

export default Main;
