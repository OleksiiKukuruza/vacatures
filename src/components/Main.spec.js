import React from 'react';
import { shallow } from 'enzyme';

import Main from './Main';

const setup = (props) => {
  const wrapper = shallow(<Main {...props} />);

  return {
    wrapper
  };
};

describe('Main', () => {
  it('renders only product list if nothing selected', () => {
    const { wrapper } = setup({ selectedVacancyId: null });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders product list if product selected', () => {
    const { wrapper } = setup({ selectedVacancyId: '1' });
    expect(wrapper).toMatchSnapshot();
  });
});
