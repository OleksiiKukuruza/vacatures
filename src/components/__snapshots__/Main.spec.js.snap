// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders without crashing 1`] = `
<styled.div>
  <ProductList />
  <Order />
</styled.div>
`;
